### ISSUE TITLE
Webpack 5 treats reexported function as unused export and replace it with `undefined`

### ISSUE THREAD

**Issue by forehalo** [webpack/webpack#11662]
# Bug report

**What is the current behavior?**
Webpack 5 treats reexported function as unused export and replace it with `undefined`, but it is used actually.

**If the current behavior is a bug, please provide the steps to reproduce.**

[reproduction link](https://codesandbox.io/s/ecstatic-meitner-c0ri4?file=/dist/main.f7047f33.js)

directly scroll down to the very bottom of the output file, you would see:

```ts
console.log(createTheme_createTheme({}));
console.log(/* unused export */ undefined({}));
```
and the corresponding input is:
```ts
// '@fluentui/react' reexports `createTheme` from '@fluentui/styling', who reexports it from '@fluentui/theme as well'.
import { createTheme as wrongCreateTheme } from "@fluentui/react";
import { createTheme as correctCreateTheme } from "@fluentui/theme";

console.log(correctCreateTheme({}));
console.log(wrongCreateTheme({}));
```

**What is the expected behavior?**
Do not take reexports wrong.

**Other relevant information:**
webpack version:  5.0.0
Node.js version: v10.21.0
Operating System: macOS 10.15.6
Additional tools: none

**Comment by sokra** [comment]
sounds like a critical bug... investigating...

**Comment by sokra** [comment]
released 5.1.0 with this fix.
